plugins/Observability/src/test/report-generator-for-regression-testing.ssc

```diff
@@ -0,0 +1,63 @@
+// Name: Observability Plugin Report Generator For Regression Testing
+// License: Public Domain
+// Author: colossatr0n
+// Version: 1.0
+// Description: Generates reports to be used for regression testing. Run this script in two different versions of Stellarium and compare the output using some kind of differ (vim, git, diff). For macOS, file is output to "~/Library/Application Support/Stellarium/".
+
+const objectNames = [
+    "Sun",      // Sun
+    "Moon",     // Moon
+    "Jupiter",  // Solar system planet
+    "Achernar", // Interstellar object
+]
+
+// Locations from different continents
+const observerLocations = [
+    "Boise, Northern America",
+    "Brasilia, Southern America",
+    "Palestine, Northern America",
+    "Barnaul, Northern Asia",
+    "Albany, Australasia",
+    "Prague, Eastern Europe"
+]
+
+Observability.showReport(true)
+
+core.resetOutput()
+var reports = []
+for (i=0; i < observerLocations.length; i++) {
+    var location = observerLocations[i]
+    core.setObserverLocation(location, 0.0)
+    var actualLocation = core.getObserverLocation()
+    var attempts = 0
+    while (location != actualLocation) {
+        if (attempts > 100) {
```

Comment:
If there is a chance that re-reading the newly-set location does not deliver it, it may be useful to add a small core.wait(0.1) in this loop.

Reply:
Also here, please add a tiny wait in the loop. This will ensure it is solved after 1-2 runs of the wait loop.

Reply:
Resolved.